verifier/src/evidence.rs

```diff
@@ -0,0 +1,334 @@
+            .set_flags(7)
+            .set_extended_features_mask(7);
+        let time = "2023-06-08T13:43:27Z"
+            .parse::<DateTime>()
+            .expect("Failed to parse time");
+        let allowed_advisories = Advisories::new(
+            [
+                "INTEL-SA-00161",
+                "INTEL-SA-00219",
+                "INTEL-SA-00289",
+                "INTEL-SA-00334",
+                "INTEL-SA-00614",
+                "INTEL-SA-00615",
+                "INTEL-SA-00617",
+            ],
+            AdvisoryStatus::OutOfDate,
+        );
+        let verifier = And::new(
+            Quote3Verifier::new(quote_key),
+            And::new(
+                MrSignerVerifier::new(mr_signer.into(), product_id, isv_svn.into()),
+                And::new(
+                    AttributesVerifier::new(attributes),
+                    And::new(
+                        TcbInfoRawVerifier::new(tcb_key, time),
+                        AdvisoriesVerifier::new(allowed_advisories),
+                    ),
+                ),
+            ),
+        );
+        verifier
+    }
+
+    #[test]
+    fn evidence_verifies_correctly() {
+        let quote_bytes = include_bytes!("../data/tests/hw_quote.dat");
+        let quote = Quote3::try_from(quote_bytes.as_ref()).expect("Failed to parse quote");
+
+        let tcb_json = include_str!("../data/tests/fmspc_00906ED50000_2023_05_10.json");
+        let tcb_info_raw = TcbInfoRaw::try_from(tcb_json).expect("Failed to parse TCB info");
+
+        let verifier = verifier(0.into(), &quote);
+        let evidence = Evidence::new(quote, tcb_info_raw).expect("Failed to create evidence");
+        let verification = verifier.verify(&evidence);
+
+        assert_eq!(verification.is_success().unwrap_u8(), 1);
+
+        let displayable = VerificationTreeDisplay::new(&verifier, verification);
+        let expected = r#"
+            - [x] Both of the following must be true:
+              - [x] The quote was signed with the provided key
+              - [x] Both of the following must be true:
+                - [x] MRSIGNER all of the following must be true:
+                  - [x] The MRSIGNER key hash should be 0xD412_A4F0_7EF8_3892_A591_5FB2_AB58_4BE3_1E18_6E5A_4F95_AB5F_6950_FD4E_B869_4D7B
+                  - [x] The ISV product ID should be 0
+                  - [x] The ISV SVN should be at least 0
+                - [x] Both of the following must be true:
+                  - [x] The attributes should be Flags: INITTED | DEBUG | MODE64BIT Xfrm: LEGACY | AVX
+                  - [x] Both of the following must be true:
+                    - [x] The raw TCB info was verified for the provided key
+                    - [x] The allowed advisories are IDs: {"INTEL-SA-00161", "INTEL-SA-00219", "INTEL-SA-00289", "INTEL-SA-00334", "INTEL-SA-00614", "INTEL-SA-00615", "INTEL-SA-00617"} Status: OutOfDate"#;
+        assert_eq!(format!("\n{displayable}"), textwrap::dedent(expected));
```

Comment:
This probably isn't necessary but I thought it was nice to see, especially in the error case

I'm fine to remove if desired.